src/core/packet.lua

```diff
@@ -109,3 +114,21 @@ function preallocate_step()
    packet_allocation_step = 2 * packet_allocation_step
 end
 
+function dump(p, w)
+   w = w or io.write
+   for i = 0, p.length-1 do
+      if i % 16 == 0 then w('\n', bit.tohex(i, -4), ': ') end
+      w(bit.tohex(p.data[i], -2), ' ')
+   end
+   w('\n')
+end
+
+ffi.metatype(packet_t, {__index = {
+   clone = clone,
+   append = append,
+   prepend = prepend,
+   shiftleft = shiftleft,
+   free = free,
+   physical = physical,
+   dump = dump,
+}})
```

Comment:
Yes, this is unrelated.  It's the kind of things that have been in my fork for a long time and I found indispensable.  sometimes I forget to backport to `master` style before PR'ing... or maybe I'm hopeful that it would finally be merged before this one.

The `:dump()` method was there mostly for debugging.  should be removed from the PR.